src/App.tsx

```diff
@@ -1,11 +1,23 @@
+import ContactDetails from './components/ContactDetails';
+import ContactList from './components/ContactList';
+import CreateContactForm from './components/CreateContactForm';
 import Home from './components/Home'
+import { Routes, Route} from "react-router-dom";
+import UpdateContactForm from './components/UpdateContactForm';
 
 
 function App() {
 
   return (
     <>
-      <Home />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/contacts" element={<ContactList />} />
+        <Route path="/contacts/:id" element={<ContactDetails />} />
+        <Route path="/create" element={<CreateContactForm />} />
+        <Route path="/contacts/:id/update" element={<UpdateContactForm />} />
```

Comment:
code style nit: recommend explicit ID naming to avoid name confusion

```suggestion
        <Route path="/contacts/:contactID" element={<ContactDetails />} />
        <Route path="/create" element={<CreateContactForm />} />
        <Route path="/contacts/:contactID/update" element={<UpdateContactForm />} />
```